src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -959,93 +987,82 @@ private AnonymousContentAccessCertificate createAnonContentAccessCertificate(
+     * @return a map that contains the unique content IDs mapped to a ProductContent instance (which embeds
+     * the enabled/disabled value in it).
      */
-    private List<ContentInfo> getContentInfo(Collection<ProductInfo> prodInfo) {
+    private Map<String, ProductContent> translateProductInfo(Collection<ProductInfo> prodInfo) {
+        Function<ProductContent, String> cidFetcher = (pcinfo) -> pcinfo.getContent().getId();
+
         return prodInfo.stream()
             .filter(Objects::nonNull)
             .map(ProductInfo::getProductContent)
             .flatMap(Collection::stream)
             .filter(Objects::nonNull)
-            .map(ProductContentInfo::getContent)
-            .toList();
+            .map(this::translateProductContentInfo)
+            .collect(Collectors.toMap(cidFetcher, Function.identity(), (v1, v2) ->
+                new ProductContent()
+                    .setContent(v2.getContent()) // last in wins
+                    .setEnabled(v1.isEnabled() || v2.isEnabled())));
     }
 
-    /**
-     * Converts {@link ContentInfo} into {@link Content}
-     *
-     * @param contentInfo
-     *  the content to convert
-     *
-     * @return the converted {@link ContentInfo} objects
-     */
-    private List<Content> convertContentInfoToContentDto(Collection<ContentInfo> contentInfo) {
-        return contentInfo.stream()
-            .filter(Objects::nonNull)
-            .map(content -> {
-                Content converted = new Content();
-                converted.setId(content.getId());
-                converted.setName(content.getName());
-                converted.setType(content.getType());
-                converted.setLabel(content.getLabel());
-                converted.setVendor(content.getVendor());
-                converted.setGpgUrl(content.getGpgUrl());
-                converted.setMetadataExpiration(content.getMetadataExpiration());
-                converted.setRequiredTags(Util.toList(content.getRequiredTags()));
-                converted.setArches(new ArrayList<>(Arch.parseArches(content.getArches())));
-                converted.setPath(content.getContentUrl());
-
-                return converted;
-            })
-            .toList();
+    private ProductContent translateProductContentInfo(ProductContentInfo pcinfo) {
+        ContentInfo cinfo = pcinfo.getContent();
+
+        org.candlepin.model.Content converted = new org.candlepin.model.Content()
+            .setId(cinfo.getId())
+            .setName(cinfo.getName())
+            .setType(cinfo.getType())
+            .setLabel(cinfo.getLabel())
+            .setVendor(cinfo.getVendor())
+            .setGpgUrl(cinfo.getGpgUrl())
+            .setMetadataExpiration(cinfo.getMetadataExpiration())
+            .setRequiredTags(cinfo.getRequiredTags())
+            .setArches(cinfo.getArches())
+            .setContentUrl(cinfo.getContentUrl());
+
+        ProductContent productContent = new ProductContent();
```

Comment:
nit: same deal here. All of this can be condensed down to an inline build:

```
return new ProductContent()
  .setContent(converted)
  .setEnabled(pcinfo.isEnabled());
```

Reply:
Updated